guides/developer/extending/integrate-your-CI/index.md

```diff
@@ -0,0 +1,126 @@
+---
+layout: single
+title:  "Integrate your CI"
+sidebar:
+  nav: guides
+---
+
+{% include toc %}
+
+At Netflix, we built CI integrations to enable features like CI in Spinnaker, fetch artifact metadata and more. You can see the full list of CI features [here](/guides/user/managed-delivery/CI-features).
+
+This guide will explain how to add an integration to your own CI provider, so you'll be able to enjoy the new features we provide.
+
+## CI view in Spinnaker
+
+We recently added the option to see CI details in Spinnaker, in a new `Builds` tab.
+It looks like this:
+{%
+  include
+  figure
+  image_path="./ci-view.png"
+%}
+
+This is a new tab that can be accessed from the main Spinnaker UI.
+The UI code (in deck) is calling `gate` service to fetch information, and `gate` is just calling `igor`.
+
+If you wish to use it, this is what you'll need to do:
+0. Make sure you are using [gate](https://github.com/spinnaker/gate). Here's the [CiController](https://github.com/spinnaker/gate/blob/master/gate-web/src/main/groovy/com/netflix/spinnaker/gate/controllers/CiController.java) the UI is calling (no need to do any actions here).
+1. Create a fork to [igor](https://github.com/spinnaker/igor)
+2. Create a new service which implements the [CiBuildService](https://github.com/spinnaker/igor/blob/master/igor-web/src/main/java/com/netflix/spinnaker/igor/ci/CiBuildService.java), and contians two endpoints:
+- `ci/builds` which returns the `GenericBuild` object.
+- `ci/builds/{buildId}/output` which returns the build's log by providing a build number, in a map format:
```

Comment:
This doesn't make sense, as that interface and implementing classes are not controllers. These should be the names of the methods you need to implement.

Reply:
These are the methods needs to be implemented. They are in a service class and not an interface.

Reply:
These are not method signatures, though. They are URI paths...